src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -172,9 +172,23 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode> {
 
   /**
    * @return The functional dependencies valid for this node. See functional_dependency.hpp for documentation.
-   *         By default, functional dependencies from both sides are forwarded. Nodes may override this behavior.
+   *         They are collected as follows:
+   *          (1) FDs are derived from the current node's unique constraints.
+   *          (2) FDs are provided by previous nodes by passing them up the LQP tree.
    */
-  virtual std::vector<FunctionalDependency> functional_dependencies() const;
+  std::vector<FunctionalDependency> functional_dependencies() const;
+
+  /**
+   * Helper function that passes FDs to upper nodes. By default, it passes FDs from the left input node.
+   * Some nodes may override this function to
+   *  - add additional FDs or
```

Comment:
additional FDs that are not based on unique constraints (e.g., `a -> a+1`, which is not implemented yet)